source/plugins/go/src/oms.go

```diff
@@ -746,70 +860,144 @@ func PostTelegrafMetricsToLA(telegrafRecords []map[interface{}]interface{}) int
 		message := fmt.Sprintf("PostTelegrafMetricsToLA::Info:derived %v metrics from %v timeseries", len(laMetrics), len(telegrafRecords))
 		Log(message)
 	}
+	
+	if IsWindows == false { //for linux, mdsd route
+		var msgPackEntries []MsgPackEntry			
+		var i int
+		start := time.Now()
+        var elapsed time.Duration
+
+		for i = 0; i < len(laMetrics); i++ { 
+				var interfaceMap map[string]interface{} 
+				stringMap := make(map[string]string)
+				jsonBytes, err := json.Marshal(*laMetrics[i])
+				if err != nil {
+					message := fmt.Sprintf("PostTelegrafMetricsToLA::Error:when marshalling json %q", err)
+					Log(message)
+					SendException(message)
+					return output.FLB_OK
+				} else {
+					if err := json.Unmarshal(jsonBytes, &interfaceMap); err != nil { 							
+						message := fmt.Sprintf("Error while UnMarshalling json bytes to interfaceMap: %s", err.Error())
+						Log(message)
+						SendException(message)
+						return output.FLB_OK
+					} else {	
+						for key, value := range interfaceMap {
+							strKey := fmt.Sprintf("%v", key)
+							strValue := fmt.Sprintf("%v", value)
+							stringMap[strKey] = strValue
+						}				
+						msgPackEntry := MsgPackEntry{							
+							Record: stringMap,
+						}
+				     	msgPackEntries = append(msgPackEntries, msgPackEntry) 
+					}				
+				}
+		}
+		if (len(msgPackEntries) > 0) {							
+				msgpBytes := convertMsgPackEntriesToMsgpBytes(MdsdInsightsMetricsTagName, msgPackEntries)			
+				if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
+					Log("Error::mdsd::mdsd connection does not exist. re-connecting ...")
+					CreateMDSDClientInsightsMetrics(ContainerType)
+					if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
+						Log("Error::mdsd::Unable to create mdsd client for insights metrics. Please check error log.")					
+						ContainerLogTelemetryMutex.Lock()
```

Comment:
guess you got these from containerlog. Mutexes need to be different for each type.

Reply:
reviewed & looks good to me. we already use the ContainerLogTelemetryMutex for both Telegraf & containerlog related to telemetry. using the same pattern. let me know if I am missing anything which could potential cause deadlock.